enterprise_subsidy/apps/transaction/management/commands/write_reversals_from_enterprise_unenrollments.py

```diff
@@ -0,0 +1,182 @@
+                    f"No existing Reversal found for unenrollment: {fulfillment_uuid}. "
+                    f"Writing Reversal for Transaction: {related_transaction}."
+                )
+
+                # On initial release we are only supporting learner initiated unenrollments for OCM courses.
+                # OCM courses are identified by the lack of an external_reference on the Transaction object.
+                # Externally referenced transactions can be unenrolled through the Django admin actions related to the
+                # Transaction model.
+                if not related_transaction.external_reference.exists():
+                    # Memoize the content metadata for the course run fetched from the enterprise catalog
+                    if not fetched_content_metadata.get(enrollment_course_run_key):
+                        try:
+                            customer_uuid = related_transaction.ledger.subsidy.enterprise_customer_uuid
+                        except Subsidy.DoesNotExist:
+                            logger.warning(
+                                f"Unable to determine enterprise customer uuid for transaction: {related_transaction}"
+                            )
+                            continue
+
+                        content_metadata = enterprise_catalog_api_client.get_content_metadata_for_customer(
+                            customer_uuid,
+                            enrollment_course_run_key,
+                        )
+                        fetched_content_metadata[enrollment_course_run_key] = content_metadata
+                    else:
+                        content_metadata = fetched_content_metadata.get(enrollment_course_run_key)
+
+                    if self.determine_refundability_of_unenrollment(content_metadata, enterprise_course_enrollment):
+                        logger.info(
+                            f"Course run: {enrollment_course_run_key} is refundable for enterprise customer user: "
+                            f"{enterprise_course_enrollment.get('enterprise_customer_user')}. Writing Reversal record."
+                        )
+
+                        # Do we need to write any additional metadata to the Reversal?
+                        reverse_full_transaction(
+                            transaction=related_transaction,
+                            idempotency_key=f'unenrollment-reversal-{fulfillment_uuid}-{enrollment_unenrolled_at}',
+                        )
+                        reversals_processed += 1
+                    else:
+                        logger.info(
+                            f"Unenrollment from course: {enrollment_course_run_key} by user: "
+                            f"{enterprise_course_enrollment.get('enterprise_customer_user')} is not refundable."
+                        )
+                else:
+                    logger.info("Found unenrollment related to an externally referenced transaction. Skipping.")
```

Comment:
Nit: for completeness, this log message should probably contain the `fulfillment_uuid` since every other branch does already.